cmd/fetch.go

```diff
@@ -0,0 +1,94 @@
+		if debug {
+			opts.Level = slog.LevelDebug
+		}
+		logger := slog.New(slog.NewTextHandler(os.Stderr, opts))
+		timing := internal.TimingConfig{
+			ImageListTimeout:          imageListTimeout,
+			InitialPullAttemptTimeout: initialPullAttemptTimeout,
+			MaxPullAttemptTimeout:     maxPullAttemptTimeout,
+			OverallTimeout:            overallTimeout,
+			InitialPullAttemptDelay:   initialPullAttemptDelay,
+			MaxPullAttemptDelay:       maxPullAttemptDelay,
+		}
+		imageList, err := loadImageNamesFromFile(imageListFile)
+		if err != nil {
+			return err
+		}
+		imageList = append(imageList, args...)
+		return internal.Run(logger, criSocket, dockerConfigJSONPath, timing, imageList...)
+	},
+}
+
+var (
+	criSocket                 string
+	dockerConfigJSONPath      string
+	imageListFile             string
+	debug                     bool
+	imageListTimeout          = time.Minute
+	initialPullAttemptTimeout = 30 * time.Second
+	maxPullAttemptTimeout     = 5 * time.Minute
+	overallTimeout            = 20 * time.Minute
+	initialPullAttemptDelay   = time.Second
+	maxPullAttemptDelay       = 10 * time.Minute
+)
+
+func init() {
+	rootCmd.AddCommand(fetchCmd)
+
+	fetchCmd.Flags().StringVar(&criSocket, "cri-socket", "/run/containerd/containerd.sock", "Path to CRI UNIX socket.")
+	fetchCmd.Flags().StringVar(&dockerConfigJSONPath, "docker-config", "", "Path to docker config json file.")
+	fetchCmd.Flags().StringVar(&imageListFile, "image-list-file", "", "Path to text file containing images to pull (one per line).")
+	fetchCmd.Flags().BoolVar(&debug, "debug", false, "Whether to enable debug logging.")
+
+	fetchCmd.Flags().DurationVar(&imageListTimeout, "image-list-timeout", imageListTimeout, "Timeout for image list calls (for debugging).")
+	fetchCmd.Flags().DurationVar(&initialPullAttemptTimeout, "initial-pull-attempt-timeout", initialPullAttemptTimeout, "Timeout for initial image pull call. Each subsequent attempt doubles it until max.")
+	fetchCmd.Flags().DurationVar(&maxPullAttemptTimeout, "max-pull-attempt-timeout", maxPullAttemptTimeout, "Maximum timeout for image pull call.")
+	fetchCmd.Flags().DurationVar(&overallTimeout, "overall-timeout", overallTimeout, "Overall timeout for a single run.")
+	fetchCmd.Flags().DurationVar(&initialPullAttemptDelay, "initial-pull-attempt-delay", initialPullAttemptDelay, "Timeout for initial delay between pulls of the same image. Each subsequent attempt doubles it until max.")
+	fetchCmd.Flags().DurationVar(&maxPullAttemptDelay, "max-pull-attempt-delay", maxPullAttemptDelay, "Maximum delay between pulls of the same image.")
+}
+
+func loadImageNamesFromFile(fileName string) ([]string, error) {
+	if fileName == "" {
+		return nil, nil
+	}
+	bytes, err := os.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	return parseImageNames(bytes), nil
+}
+
+func parseImageNames(bytes []byte) []string {
+	var imageNames []string
+	for _, line := range strings.Split(string(bytes), "\n") {
```

Comment:
nit: consider using scanner https://pkg.go.dev/bufio#NewScanner

Reply:
I did consider it but it seemed to provide no benefits since we load everything into memory anyway...